e2e/e2e.go

```diff
@@ -152,6 +147,22 @@ func grabDependencies(ctx context.Context, deps map[string]bool, name string, au
 	return nil
 }
 
+func parseServiceYaml(_ context.Context) (*Service, error) {
+	f, err := os.Open("service.yaml")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read service.yaml")
+	}
+	defer f.Close()
+
+	var s *Service
+	err = yaml.NewDecoder(f).Decode(&s)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to parse service.yaml")
+	}
```

Comment:
nit: Generally don't decode into a pointer, reference it later where needed (which would be the return)

Second part is always try to combine `if err := whatever(); err != nil` to make it easier to handle errors

```suggestion
	var s Service
	if 	err := yaml.NewDecoder(f).Decode(&s); err != nil {
		return nil, errors.Wrapf(err, "failed to parse service.yaml")
	}
```

Reply:
https://github.com/getoutreach/devbase/pull/193